openmc/geometry.py

```diff
@@ -127,34 +123,52 @@ def export_to_xml(self, path='geometry.xml', remove_surfs=False):
             self.remove_redundant_surfaces()
 
         # Create XML representation
-        root_element = ET.Element("geometry")
-        self.root_universe.create_xml_subelement(root_element, memo=set())
+        element = ET.Element("geometry")
+        self.root_universe.create_xml_subelement(element, memo=set())
 
         # Sort the elements in the file
-        root_element[:] = sorted(root_element, key=lambda x: (
+        element[:] = sorted(element, key=lambda x: (
             x.tag, int(x.get('id'))))
 
         # Clean the indentation in the file to be user-readable
-        xml.clean_indentation(root_element)
+        xml.clean_indentation(element)
+        xml.reorder_attributes(element)  # TODO: Remove when support is Python 3.8+
+
+        return element
+
+    def export_to_xml(self, path='geometry.xml', remove_surfs=False):
+        """Export geometry to an XML file.
+
+        Parameters
+        ----------
+        path : str
+            Path to file to write. Defaults to 'geometry.xml'.
+        remove_surfs : bool
+            Whether or not to remove redundant surfaces from the geometry when
+            exporting
+
+            .. versionadded:: 0.12
+
+        """
+        root_element = self.to_xml_element(remove_surfs)
 
         # Check if path is a directory
         p = Path(path)
         if p.is_dir():
             p /= 'geometry.xml'
 
         # Write the XML Tree to the geometry.xml file
-        xml.reorder_attributes(root_element)  # TODO: Remove when support is Python 3.8+
         tree = ET.ElementTree(root_element)
         tree.write(str(p), xml_declaration=True, encoding='utf-8')
 
     @classmethod
-    def from_xml(cls, path='geometry.xml', materials=None):
-        """Generate geometry from XML file
+    def from_xml_element(cls, elem, materials=None):
+        """Generate geometry from an XML element
 
         Parameters
         ----------
-        path : str, optional
-            Path to geometry XML file
+        elem : xml.etree.ElementTree.Element
+            XML element
         materials : openmc.Materials or None
             Materials used to assign to cells. If None, an attempt is made to
             generate it from the materials.xml file.
```

Comment:
Now it looks like this description needs to be updated (a dictionary is being passed)